### PR TITLE
Recover tenant overrides when submission fails

### PR DESCRIPTION
**Pull request by fuhuifang**
Add a `try/except` block to make sure overrides are reverted when `build_and_submit_new_pipeline()` raises an exception.

New stdout:

[2021-11-04 23:28:37,203][shrike.pipeline.pipeline_helper][ERROR] - An error UserErrorException:
        Message: Cannot find dataset registered with name "data_doesnt_exist" (version: None) in the workspace.
        InnerException None
        ErrorResponse
{
    "error": {
        "code": "UserError",
        "message": "Cannot find dataset registered with name \"data_doesnt_exist\" (version: None) in the workspace."
    }
} occurred during pipeline submission.
**[2021-11-04 23:28:37,203][shrike.pipeline.pipeline_helper][INFO] - Now trying to recover overrides.
[2021-11-04 23:28:37,203][shrike.pipeline.pipeline_helper][INFO] - Reverting changes to spec yaml files. Keeping modified spec yaml files: False.**

### CODE REVIEW

**Comment by fuhuifang** [comment]
This change is pretty trivial, is unit test required? I've verified locally.

File: shrike/pipeline/pipeline_helper.py [hunk 1]
```diff
@@ -1467,15 +1476,18 @@ def run(self):
...
-                    log.error(f"An error occured, recovery is not successful: {e}")
+                        log.error(f"An error occurred, override is not successful: {e}")
+            try:
+                pipeline_run = self.build_and_submit_new_pipeline()
+            except BaseException as e:
+                log.error(f"An error {e} occurred during pipeline submission.")
+                if override:
+                    log.info("Now trying to recover overrides.")
+                self._recover_tenant_overrides(override, yaml_to_be_recovered, False)
+            else:
```
Comment (haozhestat) [inline shrike/pipeline/pipeline_helper.py:1487]: I was wondering whether the indent of the `else` statement is wrong?

File: shrike/pipeline/pipeline_helper.py [hunk 2]
```diff
@@ -1345,6 +1345,15 @@ def _recover_spec_yaml(self, spec_file_pairs, keep_modified_files):
                     old_env_file_path, os.path.splitext(old_env_file_path)[0] + ".yaml"
                 )
 
+    def _recover_tenant_overrides(
```
Comment (haozhestat) [inline shrike/pipeline/pipeline_helper.py:1348]: Could you add a unit test for this function? (It is optional.)

File: shrike/pipeline/pipeline_helper.py (hunk 1 above)
Comment (fuhuifang) [reply shrike/pipeline/pipeline_helper.py:1487]: Yes, the `else` block is designed to recover files if no exception is raised.

File: shrike/pipeline/pipeline_helper.py (hunk 2 above)
Comment (fuhuifang) [reply shrike/pipeline/pipeline_helper.py:1348]: I don't think this unit test is quite necessary, as it's just calling `_recover_spec_yaml()` which has been unit tested.

**Review by haozhestat — APPROVED** [review]

**Pull request merged**